### PR TITLE
Fix a bug causes wallet lockup when making transactions

### PR DESCRIPTION
**Pull request by cjepson**
A bug existed that could cause the unusual locking up of the wallet
when requesting a transaction, such as through sendtoaddress. The
new function MakeInputSource opens a database transaction, but when
the address pool runs out of addresses it also attempts to open a
database transaction through waddrmgr.NextInternalAddresses. This
would cause a gridlock. To fix this, the address is pulled before
the database transaction is opened and then committed to later if
the transaction succeeds and includes the change address.

### CODE REVIEW

File: wallet/createtx.go [hunk 1]
```diff
@@ -368,25 +368,38 @@ func (w *Wallet) txToOutputs(outputs []*wire.TxOut, account uint32, minconf int3
...
 			pool.BatchFinish()
 		} else {
 			pool.BatchRollback()
 		}
 	}()
 
-	inputSource := w.TxStore.MakeInputSource(account, minconf, bs.Height)
+	// The change address is pulled here rather than after
+	// MakeInputSource is called because MakeInputSource
+	// accesses the database in a way that is unsafe from
```
Comment (jrick) [inline wallet/createtx.go]: It's not unsafe, it just deadlocks.  Just say that the two functions can use separate transactions, one an update, and so the update needs to happen first.

**Comment by jcvernaleo** [comment]
This is running fine in my testing (but since I haven't been able to reproduce the bug today that doesn't mean too much).  @alexlyp might have more useful input.

**Comment by alexlyp** [comment]
tACK from me.  I was able to run 6 tests cleanly then the first attempt back on master locked up pretty quickly.  Appears this fixes the problem

**Comment by jcvernaleo** [comment]
I think this just needs a squash and rebase now.

**Pull request merged**